src/application/jobs/notifier-rappel-instance-session-milo.job.handler.ts

```diff
@@ -0,0 +1,93 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { JobHandler } from '../../building-blocks/types/job-handler'
+import { isSuccess } from '../../building-blocks/types/result'
+import {
+  JeuneMilo,
+  MiloJeuneRepositoryToken
+} from '../../domain/milo/jeune.milo'
+import { RendezVousMilo } from '../../domain/milo/rendez-vous.milo'
+import {
+  SessionMilo,
+  SessionMiloRepositoryToken
+} from '../../domain/milo/session.milo'
+import {
+  Notification,
+  NotificationRepositoryToken
+} from '../../domain/notification/notification'
+import { Planificateur, ProcessJobType } from '../../domain/planificateur'
+import { SuiviJob, SuiviJobServiceToken } from '../../domain/suivi-job'
+import { DateService } from '../../utils/date-service'
+
+interface Stats {
+  idDossier: string
+  notificationEnvoyee: boolean
+}
+
+@Injectable()
+@ProcessJobType(Planificateur.JobType.RAPPEL_SESSION)
+export class NotifierRappelInstanceSessionMiloJobHandler extends JobHandler<
+  Planificateur.Job<Planificateur.JobRappelSession>
+> {
+  constructor(
+    @Inject(SuiviJobServiceToken)
+    suiviJobService: SuiviJob.Service,
+    @Inject(SessionMiloRepositoryToken)
+    private sessionMiloRepository: SessionMilo.Repository,
+    @Inject(MiloJeuneRepositoryToken)
+    private jeuneRepository: JeuneMilo.Repository,
+    @Inject(NotificationRepositoryToken)
+    private notificationRepository: Notification.Repository,
+    private dateService: DateService
+  ) {
+    super(Planificateur.JobType.RAPPEL_SESSION, suiviJobService)
+  }
+
+  async handle(
+    job: Planificateur.Job<Planificateur.JobRappelSession>
+  ): Promise<SuiviJob> {
+    const debut = this.dateService.now()
+
+    const instance = await this.sessionMiloRepository.findInstanceSession(
+      job.contenu.idInstance,
+      job.contenu.idDossier
+    )
+
+    const stats: Stats = {
+      idDossier: job.contenu.idDossier,
+      notificationEnvoyee: false
+    }
+
+    if (instance) {
+      const resultJeune = await this.jeuneRepository.getByIdDossier(
+        job.contenu.idDossier
+      )
+      if (isSuccess(resultJeune)) {
+        if (resultJeune.data.configuration?.pushNotificationToken) {
+          const dateTimezonee = RendezVousMilo.timezonerDateMilo(
+            instance.dateHeureDebut,
+            resultJeune.data
+          )
+
+          const notification = Notification.creerNotificationRappelSessionMilo(
+            resultJeune.data.configuration.pushNotificationToken,
+            job.contenu.idSession,
+            dateTimezonee,
+            this.dateService
+          )
+          if (notification) {
+            await this.notificationRepository.send(notification)
+          }
+          stats.notificationEnvoyee = true
```

Comment:
XXXX Probleme : actuellement la valeur est affectée  a true dans tout les cas 

-mettre le stats.notificationEnvoyee dans le if
-affecter this.notificationRepository.send a une variable result 
-> comme ça on passe la valeur a true que si on a un success sur le result

Reply:
yes merci